nomad/host_volume_endpoint.go

```diff
@@ -610,6 +610,7 @@ func (v *HostVolume) deleteVolume(vol *structs.HostVolume) error {
 	method := "ClientHostVolume.Delete"
 	cReq := &cstructs.ClientHostVolumeDeleteRequest{
 		ID:         vol.ID,
+		Name:       vol.Name,
```

Comment:
Maybe I'm missing it, but I don't see what's changed in the client to necessitate adding this field here?

Reply:
this ends up down in `HostVolumeManager.Delete` so it can be passed into `updateNodeVols` here: https://github.com/hashicorp/nomad/pull/24613/files#diff-0ca03dbe7624ffebd44da8e738ac49a768b68a96c4f612f24f4f0bb606b81c65R214

in fact, I was struggling to understand why the addition to the endpoint test was failing, until I realized I needed this:
https://github.com/hashicorp/nomad/pull/24613/files#diff-054b6367d0da40a926f9565f6129bf0fabc62f7c2bb154540a8c69ac5f10b5cdR74
because otherwise `""` was being "deleted" from the map, leaving the actual volume in place.